springwolf-plugins/springwolf-cloud-stream-plugin/src/main/java/io/github/springwolf/plugins/cloudstream/annotation/GooglePubSubSchemaSetting.java

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: Apache-2.0
+package io.github.springwolf.plugins.cloudstream.annotation;
+
+import io.github.springwolf.asyncapi.v3.bindings.googlepubsub.GooglePubSubSchemaSettings;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target(value = {ElementType.METHOD})
+@Inherited
+public @interface GooglePubSubSchemaSetting {
```

Comment:
I'd like to keep protocol-specific annotations out of core and cloudstream.

@ctasada is working https://github.com/springwolf/springwolf-core/pull/634 to introduce an own gradle module for each binding.
This seems highly suitable for the google pubsub bindings as well.
There the `@AsyncGooglePubSubChannelBinding` and `AsyncGooglePubSubSchema` annotation can be put together.

This has the benefit that also other plugins can use the pubsub binding and also the cloudstream and pubsub is not intertwined.

Reply:
Yes, I believe this will provide us with an opportunity to redesign the `getProtocolName()` method as well. Rather solely relying on extracting protocol information from the first server definition via `docket.getServers()`, we can also consider setting the protocol based on annotations. This approach allows us to specify various protocols for different methods.

Reply:
@SheheryarAamir I just merged https://github.com/springwolf/springwolf-core/pull/655 I think you can follow the a similar approach for this binding.

Reply:
sure, thank you